ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/drawing-and-display.tsx

```diff
@@ -13,18 +13,219 @@
  *
  **/
 import React from 'react'
+import { useState, useEffect } from 'react'
 import {
   getDrawModeFromModel,
   useDrawingAndDisplayModels,
+  DrawModeType,
 } from '../drawing-and-display'
 import { OpenlayersBboxDisplay } from './bbox-display'
-import { OpenlayersBboxDrawing } from './bbox-drawing'
 import { OpenlayersCircleDisplay } from './circle-display'
-import { OpenlayersCircleDrawing } from './circle-drawing'
 import { OpenlayersLineDisplay } from './line-display'
-import { OpenlayersLineDrawing } from './line-drawing'
 import { OpenlayersPolygonDisplay } from './polygon-display'
-import { OpenlayersPolygonDrawing } from './polygon-drawing'
+import styled from 'styled-components'
+import { menu, geometry } from 'geospatialdraw'
+import { Shape } from 'geospatialdraw/target/webapp/shape-utils'
+import {
+  GeometryJSON,
+  makeGeometry,
+  makePointRadiusGeo,
+  makeLineGeo,
+  makeBBoxGeo,
+} from 'geospatialdraw/target/webapp/geometry'
+import { transparentize } from 'polished'
+import * as ol from 'openlayers'
+import Common from '../../../../js/Common'
+import wreqr from '../../../../js/wreqr'
+import { Drawing } from '../../../../component/singletons/drawing'
+import * as Turf from '@turf/turf'
+import { Position, Point, Polygon, LineString } from '@turf/turf'
+import { validateGeo } from '../../../../react-component/utils/validation'
+import ShapeUtils from '../../../../js/ShapeUtils'
+import _ from 'lodash'
+
+// TODO should move all the common code to ddf-ui and have downstream import it
```

Comment:
✏️  Remove this TODO

Reply:
❓ should this TODO be removed?

Reply:
This is still valid and I'm addressing it now.